build.sh

```diff
@@ -1036,6 +1033,10 @@ while :; do
     shift
 done
 
+if [ "${__BuildArch}" != "${__HostArch}" ]; then
```

Comment:
This doesn't have to be true. If you install cross-arch libraries on ARM64, you can build for ARM without cross building. Similar to building for x86 on x64. So I'd prefer keeping it as it was.

Reply:
If we keep this as is, we effectively have to recalculate the host architecture in the MSBuild project by copying the script from build.sh and hooking it into MSBuild, which adds a significant amount of complexity.

Additionally, we don't currently support building via cross-arch libraries in our build script. The current implementation in master in all cases where we will now automatically cross-build would previously error out and say "pass the -cross parameter". It would never use the cross-arch libraries in the ARM-on-ARM64 or x86-on-x64 cases.

Reply:
This change effectively removes the dev workflow of running the build script, having the build script say "you need to pass the cross parameter", and then passing the cross parameter to get the build to run by just doing the cross build. If we want to add more smarts to the build script, we should add that here later.

Reply:
Ok, then keep it. It used to work in the past, but I can see that the message and check was added about 9 months ago, so probably noone cares.